cmake/process_users_input.cmake

```diff
@@ -140,6 +139,23 @@ else()
 endif()
 message(STATUS "GEMM TOOLS are: ${GEMM_TOOLS_LIST}")
 
+
+if (DEVICE_ARCH MATCHES "sm_*")
+    set(DEVICE_VENDOR "nvidia")
+    set(DEVICE_FLUXMERGE_DEFAULT ON)
+elseif(DEVICE_ARCH MATCHES "gfx*")
+    set(DEVICE_VENDOR "amd")
+    set(DEVICE_FLUXMERGE_DEFAULT ON)
+else()
+    # TODO(David): adjust as soon as we add support for more vendors
+    set(DEVICE_VENDOR "intel")
+    set(DEVICE_FLUXMERGE_DEFAULT OFF)
+endif()
+
+if (WITH_GPU)
+    option(DEVICE_FLUXMERGE "Merge device flux matrices (recommended for AMD and Nvidia GPUs)" ${DEVICE_FLUXMERGE_DEFAULT})
```

Comment:
Could you rename `DEVICE_FLUXMERGE` to something like `MERGE_FLUX` or `MERGED_FLUX`. Yes, it affects GPU performance. But, technically speaking, it belongs to numerics. I mean we don't need to be super specific and add `DEVICE` prefix. 

When it says `DEVICE` I immediately think that it belongs to the Device API (device submodule). However, the changes affect mainly YATeTo high level description of the numerics.

If you decide to go for the split variant you can name it `SPLIT_FLUX`

Reply:
Done